### PR TITLE
fixes #3416, eventDestroy not called when removing the popover

### LINKED ISSUE #3416: eventDestroy not called when removing the popover
### Bug Reports
The method to ```showSegPopover``` reuses the events and render them in the popover. 

It populates a list of ```popoverSegs```, but ```eventDestroy()``` is not called for those "new" dom elements.

```
			hide: function() {
				// kill everything when the popover is hidden
				_this.segPopover.removeElement();
				_this.segPopover = null;
				_this.popoverSegs = null;
			}
```

Therefore it creates a leak that we cannot cleanup like original eventRender / eventDestroy in normal fullcalendar.

### Feature Requests
In hide function. Iterate over the ```_this.popoverSegs``` and trigger ```eventDestroy()```

### PR DESCRIPTION
**Pull request by Rouche**
https://github.com/fullcalendar/fullcalendar/issues/3416

### CODE REVIEW

**Comment by Rouche** [comment]
Will fix in next release branch.

**Comment by arshaw** [comment]
Thanks! I'll get a chance to play around with these code changes as soon as I can. It would save me time if you'd be so kind as to write some tests to prove that the bug case you're addressing is solved. More info: https://github.com/fullcalendar/fullcalendar/wiki/Automated-Tests

**Comment by Rouche** [comment]
Done

File: tasks/locale.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ gulp.task('locale:each:data', function() {
 	return gulp.src('node_modules/moment/locale/*.js')
 		.pipe(modify({
 			fileModifier: function(file, momentContent) {
-				var localeCode = file.path.match(/([^\/]*)\.js$/)[1];
+				var localeCode = file.path.match(/([^\/\\]*)\.js$/)[1];
```
Comment (arshaw) [inline tasks/locale.js]: why did you need to change this?

File: tasks/locale.js [hunk 2]
```diff
@@ -183,7 +183,7 @@ function extractMomentLocaleJS(js) {
 
 	// remove the UMD wrap
 	js = js.replace(
-		/\(\s*function[\S\s]*?function\s*\(\s*moment\s*\)\s*\{([\S\s]*)\}\)\);?/,
+		/\(\s*function[\S\s]*?function\s*\(\s*moment\s*\)\s*\{([\S\s]*)\}\)\)\)?;?/,
```
Comment (arshaw) [inline tasks/locale.js]: why did you need to change this?

File: tests/automated/eventLimit-popoverDestroy.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
+
+describe('eventLimit popover', function() {
```
Comment (arshaw) [inline tests/automated/eventLimit-popoverDestroy.js]: thanks!

File: src/common/DayGrid.limit.js [hunk 4]
```diff
@@ -247,6 +247,14 @@ DayGrid.mixin({
 			viewportConstrain: view.opt('popoverViewportConstrain'),
 			hide: function() {
 				// kill everything when the popover is hidden
+				// notify events to be removed
+				if(_this.popoverSegs) {
```
Comment (arshaw) [inline src/common/DayGrid.limit.js]: space after if --- `if (`

File: src/common/DayGrid.limit.js [hunk 5]
```diff
@@ -247,6 +247,14 @@ DayGrid.mixin({
...
+					var seg;
+					for (var i = 0; i < _this.popoverSegs.length; ++i) {
+						seg = _this.popoverSegs[i];
+						view.trigger('eventDestroy', seg.event, seg.event, seg.el);
+					};
```
Comment (arshaw) [inline src/common/DayGrid.limit.js]: no semicolon necessary here

File: tasks/locale.js (hunk 2 above)
Comment (Rouche) [inline tasks/locale.js]: The build is failing, Seems like MomentJS changed.

File: tasks/locale.js (hunk 1 above)
Comment (Rouche) [inline tasks/locale.js]: To support windows.

File: src/common/DayGrid.limit.js (hunk 5 above)
Comment (Rouche) [inline src/common/DayGrid.limit.js]: This is weird, lint passed. Ill correct this tomorrow morning.

**Comment by arshaw** [comment]
released in [v3.1.0](https://github.com/fullcalendar/fullcalendar/releases/tag/v3.1.0)

**Comment by arshaw** [comment]
thanks for the contribution!
